src/parse.c

```diff
@@ -5264,13 +5264,47 @@ static class_t p_entity_class(void)
    }
 }
 
+static ident_t p_entity_designator()
+{
+   // entity_tag [ signature ]
+   ident_t id = p_identifier();
+
+   if (peek() == tLSQUARE) {
+      // XXX: Review what to do here
+      type_t constraints = p_signature();
```

Comment:
I don't know what to do here with the signature.  Any suggestions?

Reply:
I would just ignoring it for now with a warning. E.g. 

```suggestion
   if (peek() == tLSQUARE) {
      // XXX: Review what to do here
      (void)p_signature();
      warn_at(CURRENT_LOC, "sorry, signature in attribute entity name list is not yet supported");
```

It needs some more refactoring to handle properly.